cpp/src/transform/compute_column.cu

```diff
@@ -88,8 +88,9 @@ std::unique_ptr<column> compute_column(table_view const& table,
   // If none of the input columns actually contain nulls, we can still use the
   // non-nullable version of the expression evaluation code path for
   // performance, so we capture that information as well.
-  auto const nullable  = cudf::nullable(table);
-  auto const has_nulls = nullable && cudf::has_nulls(table);
+  auto const expr_has_nulls = ast::detail::contains_null_literal(&expr);
+  auto const nullable       = cudf::nullable(table) || expr_has_nulls;
+  auto const has_nulls      = (nullable && cudf::has_nulls(table)) || expr_has_nulls;
```

Comment:
Your interpretation is correct, at some point the call to count the null bits in the bitmask does a `mask == nullptr` check. This was just some premature (and frankly pretty unnecessary) optimization on my part. It does require a _little_ bit more work than just function calls since you also generate iterators for the columns in the table to check them all, but you're right that the overhead will probably be negligible. The appropriate changes will depend on the discussion above, though.

Reply:
This point is no longer relevant with the new `may_evaluate_null` method.